pgjdbc/src/test/java/org/postgresql/test/jdbc2/DatabaseMetaDataTest.java

```diff
@@ -492,6 +492,23 @@ public void testDroppedColumns() throws SQLException {
     assertEquals(3, rs.getInt("ORDINAL_POSITION"));
     assertTrue(!rs.next());
     rs.close();
+
+    /* getFunctionColumns also has to be aware of dropped columns
+       add this in here to make sure it can deal with them
+     */
+    rs = dbmd.getFunctionColumns(null, null, "f2", null);
```

Comment:
Does this reproduce the problem?
I thought the function should be somehow relate to the table.
As far as I see, `f2` is defined as `f2(a int, b varchar) RETURNS int`, so I see no way it is related with `TABLE metadatatest `